bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.sddp;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A DTO class containing data from an SDDP device discovery result.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class SddpDevice {
+
+    public final String from;
+    public final String host;
+    public final String maxAge;
+    public final String type;
+    public final String primaryProxy;
+    public final String proxies;
+    public final String manufacturer;
+    public final String model;
+    public final String driver;
+    public final Instant expireInstant;
+
+    /**
+     * Constructor
+     * 
+     * @param headers a map of parameter name / value pairs.
+     */
+    public SddpDevice(Map<String, String> headers) {
+        from = headers.getOrDefault("From", "").replaceAll("^\"|\"$", "");
```

Comment:
It would also be helpful to split the **from:** field on the colon and provide the ip address as its own field. The port number here is not particularly useful since it is just the multicast port.

Sample **From:**
From: "192.168.4.237:1902"

Reply:
Yes. Ok.

Reply:
PS even though all examples have port 1902, I believe it would theoretically be possible for another value to be present, so I also added a port property just-in-cas.